### PR TITLE
auto re-import atlas, fixes #40047

### LINKED ISSUE #40047: Atlas auto re-importing
**Godot version:**
v3.2.2.stable.official

**OS/device including version:**
Windows 10

**Issue description:**
Does texture atlas update by detecting changing one of consist sprites? Expected behiavour like working with individual sprites.

**Steps to reproduce:**
1. unzip attached example.
2. open project.
3. open textures folder from project
4. open one of sprites (a.png, b.png,c.png.) with any image editor and redraw.
5. goto godot editor.
6. atlas doesn't updated.

_but you can manually update atlas by:_
a) close godot editor and reopen project.
or
b) choose atlas and hit re-import in "import" tab.

**Minimal reproduction project:**
here -> [example.zip](https://github.com/godotengine/godot/files/4862786/example.zip)

### PR DESCRIPTION
**Pull request by ChristopheLY**
Fixes #40047.

### CODE REVIEW

File: editor/editor_file_system.cpp [hunk 1]
```diff
@@ -532,7 +532,7 @@ bool EditorFileSystem::_update_scan_actions() {
 				String full_path = ia.dir->get_file_path(idx);
 				if (_test_for_reimport(full_path, false)) {
 					//must reimport
-					reimports.push_back(full_path);
+					reimports = _get_dependencies(full_path);
```
Comment (akien-mga) [inline editor/editor_file_system.cpp:535]: Doesn't this discard all the other paths which had previously been added to `reimports`?

Comment (ChristopheLY) [reply editor/editor_file_system.cpp:535]: Yes, that's a problem ...

Comment (ChristopheLY) [reply editor/editor_file_system.cpp:535]: With this it's work to and don't discard the other paths
image

File: editor/editor_file_system.cpp [hunk 2]
```diff
@@ -532,7 +532,12 @@ bool EditorFileSystem::_update_scan_actions() {
 				String full_path = ia.dir->get_file_path(idx);
 				if (_test_for_reimport(full_path, false)) {
 					//must reimport
-					reimports.push_back(full_path);
+					List<String> deps;
+					ResourceLoader::get_dependencies(full_path, &deps);
+
+					for (List<String>::Element *E = deps.front(); E; E = E->next()) {
+						reimports.push_back(E->get());
+					}
```
Comment (akien-mga) [inline editor/editor_file_system.cpp:540]: I think you could simplify with this:
```
Vector<String> deps = _get_dependencies();
reimports.append_array(deps);
```

Comment (ChristopheLY) [reply editor/editor_file_system.cpp:540]: just  `_get_dependencies` and not `_get_dependencies(full_path)` ?

Comment (akien-mga) [reply editor/editor_file_system.cpp:540]: Er yes, with `full_path`.

**Review by akien-mga — APPROVED** [review]

File: editor/editor_file_system.cpp [hunk 3]
```diff
@@ -532,7 +532,7 @@ bool EditorFileSystem::_update_scan_actions() {
 				String full_path = ia.dir->get_file_path(idx);
 				if (_test_for_reimport(full_path, false)) {
 					//must reimport
-					reimports.push_back(full_path);
+					reimports.append_array(_get_dependencies(full_path));
```
Comment (akien-mga) [inline editor/editor_file_system.cpp:535]: Actually, shouldn't we still add `full_path` itself too, instead of only its dependencies?

**Review by akien-mga — CHANGES_REQUESTED** [review]
See my comment, I tested it locally and this completely breaks import of resources...

**Comment by akien-mga** [comment]
Here's what this PR gives in a new project:
Screenshot_20200710_090439

Since `full_path` is no longer imported, all resource imports apart from AtlasTextures are broken.

**Comment by ChristopheLY** [comment]
So, we just need to readd `reimports.push_back(full_path);` to not break resources imported

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.3.

**Pull request merged**